cputree.c

```diff
@@ -366,21 +366,23 @@ static void do_one_cpu(char *path)
 		struct topo_obj *node;
 
 		dir = opendir(path);
-		do {
-			entry = readdir(dir);
-			if (!entry)
-				break;
-			if (strncmp(entry->d_name, "node", 4) == 0) {
-				char *end;
-				int num;
-				num = strtol(entry->d_name + 4, &end, 10);
-				if (!*end && num >= 0) {
-					nodeid = num;
+		if (dir) {
```

Comment:
Instead of nesting the do/while in an if, why not change this the do {}while to :
while (dir && entry) { ...} if(dir) { closedir(dir)}; 
That would save you a level of nesting, which might be nice